### DISCUSSION TITLE
Support for Dynamic Groups (i.e. @host)

### DISCUSSION THREAD

**Discussion by mikekasprzak** [ludumdare/ludumdare#2000] [Ideas]
Add support for dynamic groups. For example, the @host group's member list should be equal to who is running the current event (me or the MiniLD host).

**Comment by mikekasprzak** [comment]
Ah yes, I forgot about this.

We definitely need a proxy user for this, someone in the users slug pool so they show up when the `@name` is used. Post-type can be something other than `user`, to know it's special.

The question is how to route the notifications to them. With 2+ events going on, `@host` should only route in posts that belong to the event you're running. Hmm.

**Comment by mikekasprzak** [comment]
When you run an event, you are auto-subscribed to notifications sent to the event itself. Technically, every entity can receive notifications. But if nobody is listening, then the notifications are never heard.

So if `@host` is used, obviously detect that, then we'll have to walk the tree backwards until we hit an event. Then assign the notification to the event itself, rather than to a specific user. If we hit the root node (i.e. posts outside an event), trigger a fail?

The above also works for events run by multiple people, as they simply both need a subscription to the event to receive its `@host` notifications. 

The slug in `@names` is simply used to figure out who to send the notification to. Notifications should be assigned to the ID, and not the slug. Notifications are only sent after the initial publish of a post or comment. Appending `@names` to a post or comment after an edit will not send a notification.

**Comment by mikekasprzak** [comment]
In the same vein, may as well have an `@author` shorthand. Unlike `@host` which could be used in posts too, `@author` would primarily be used for comments.

Similar to, but rather than walking the tree, it simply assigns the notification to the post itself.

This would then mean everybody is automatically (implicitly?) subscribed to every post they make.

**Comment by local-minimum** [comment]
Isnt it a bit confusing to have @ both for at-names and for dynamic groups? What happens if we have a user named 'host'?

Suggest double at for groups: `@@host`, though not so pretty

**Comment by zwrawr** [comment]
I think it makes sense, lots of services have stuff like `@mods` or `@admins`.
Names like author, mod, host are reserved, there's a list somewhere in the repo.

**Comment by mikekasprzak** [comment]
Yes. In anticipation of this, I blacklisted a large number of words before I opened the site up to create accounts to stop people from creating accounts like @admin, @host, etc.
